examples/src/examples/misc/hello-script-modules.mjs

```diff
@@ -0,0 +1,102 @@
+import * as pc from 'playcanvas';
+
+/**
+ * @param {import('../../options.mjs').ExampleOptions} options - The example options.
+ * @returns {Promise<pc.AppBase>} The example application.
+ */
+async function example({ canvas, deviceType, glslangPath, twgslPath }) {
+
+    const gfxOptions = {
+        deviceTypes: [deviceType],
+        glslangUrl: glslangPath + 'glslang.js',
+        twgslUrl: twgslPath + 'twgsl.js'
+    };
+
+    const device = await pc.createGraphicsDevice(canvas, gfxOptions);
+    const createOptions = new pc.AppOptions();
+    createOptions.mouse = new pc.Mouse(document.body);
+    createOptions.graphicsDevice = device;
+
+    createOptions.componentSystems = [
+        // @ts-ignore
+        pc.EsmScriptComponentSystem,
+        // @ts-ignore
+        pc.RenderComponentSystem,
+        // @ts-ignore
+        pc.CameraComponentSystem,
+        // @ts-ignore
+        pc.LightComponentSystem
+
+    ];
+    createOptions.resourceHandlers = [
+        // @ts-ignore
+        pc.TextureHandler,
+        // @ts-ignore
+        pc.ContainerHandler
+    ];
+
+    const app = new pc.AppBase(canvas);
+    app.init(createOptions);
+    app.start();
+
+    // Set the canvas to fill the window and automatically change resolution to be the same as the canvas size
+    app.setCanvasFillMode(pc.FILLMODE_FILL_WINDOW);
+    app.setCanvasResolution(pc.RESOLUTION_AUTO);
+
+    // Ensure canvas is resized when window changes size
+    const resize = () => app.resizeCanvas();
+    window.addEventListener('resize', resize);
+    app.on('destroy', () => {
+        window.removeEventListener('resize', resize);
+    });
+
+    // create box entity
+    const box = new pc.Entity('World');
+    box.addComponent('render', {
+        type: 'box'
+    });
+
+    box.addComponent('esmscript');
+
+
+    // The paths are in node, not relative to the url
+    // const path = require('path');
+    console.log(`${document.location.origin}/scripts/make-confetti.js`);
+    const Confetti = await import(`${document.location.origin}/scripts/make-confetti.js`);
+    // const Confetti = await import('/examples/src/examples/misc/make-confetti.js')
+    const Rotate = await import(`${document.location.origin}/scripts/rotate.js`);
+
+    box.esmscript.add(Rotate);
+    box.esmscript.add(Confetti, {
+        confettiSettings: {
+            particleCount: 200
+        }
+    });
+
+    app.root.addChild(box);
+
```

Comment:
so here I added a code like this:
```
    const box2 = box.clone();
    box2.setLocalPosition(1, 0, 0);
    app.root.addChild(box2);

```
and the second box renders but does not rotate.

Reply:
Ok, this should be working now @mvaligursky

Reply:
OK that should be fixed now @mvaligursky